### PR TITLE
Ta 191 cohorts can be created with names 1

### PR DESCRIPTION
**Pull request by Dan2024**
https://github.com/boolean-uk/team-dev-client/issues/191

### CODE REVIEW

**Review by Alex90Jennings — APPROVED** [review]

File: src/components/cohorts/AddCohortForm.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const AddCohortForm = () => {
+  let navigate = useNavigate()
+  const [cohortFormData, setCohortFormData] = useState({ cohort_name: '' });
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    client
+      .post('/cohort', cohortFormData)
+      .then((res) => {
+        navigate(`../cohorts/${res.data.data.cohort.id}`, { replace: true })
```
Comment (phillipdunn) [inline src/components/cohorts/AddCohortForm.js:18]: You can shorten this to make it more readable and not so 'data'tastic by destructuring on the line above
`const data = res.data.data;`
then have 
```navigate(`../cohorts/${data.cohort.id}```

File: src/components/cohorts/CohortPage.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const CohortPage = () => {
+  const params = useParams();
+  const [cohort, setCohort] = useState()
+
+  useEffect(() => {
+    client
+        .get(`/cohort/${params.id}`)
+        .then((res) => {
+          console.log(res.data.data);
```
Comment (phillipdunn) [inline src/components/cohorts/CohortPage.js:15]: same here

File: src/components/cohorts/AddCohortForm.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      })
+      .catch((data) => console.log(data));
```
Comment (phillipdunn) [inline src/components/cohorts/AddCohortForm.js:20]: probably want to catch the error response here?

File: src/App.js [hunk 4]
```diff
@@ -46,6 +49,15 @@ function App() {
           >
             <Route path='/log' element={<DeliveryLogDash />} />
           </Route>
+
+          <Route element={<AuthenticateUser redirectPath={'/posts'} requiredRole={['TEACHER']}/>}>
+            <Route path='/cohorts/add-cohort' element={<AddCohortForm />} />
+          </Route>
+
+          <Route element={<AuthenticateUser redirectPath={'/posts'} requiredRole={['TEACHER']}/>}>
```
Comment (Auenc) [inline src/App.js:57]: We don't need to have two of these protected routes. I believe we should be able to add the new `add-cohort` route and the `/cohorts/:id` nested inside one `AuthenticateUser` route.

**Review by Auenc — CHANGES_REQUESTED** [review]
Just a minor change here

**Review by Alex90Jennings — APPROVED** [review]

**Review by phillipdunn — APPROVED** [review]

**Review by Alex90Jennings — APPROVED** [review]

**Review by Auenc — APPROVED** [review]
LGTM
